ion-schema/src/ordered_elements_nfa.rs

```diff
@@ -424,9 +424,9 @@ impl Display for State {
 /// The reason why a transition (or edge) in the state machine graph cannot be traversed.
 #[derive(Debug)]
 enum TraversalError {
-    CannotEnterState(StateId, Violation),
-    CannotExitState(StateId),
-    CannotReEnterState(StateId),
+    Entering(StateId, Violation),
+    Exiting(StateId),
+    ReEntering(StateId),
```

Comment:
🗺️ Clippy complained that the enum variants all had the same prefix, suggesting that it's an implied part of the variant name or should be part of the enum name. When I looked it up, it seemed like it would also complain about a commin suffix, so I shortened all of the names to be the gerund form of the attempted action rather than a complete description.